balancer/endpointsharding/endpointsharding.go

```diff
@@ -28,19 +28,30 @@ package endpointsharding
 import (
 	"encoding/json"
 	"errors"
+	rand "math/rand/v2"
 	"sync"
 	"sync/atomic"
 
-	rand "math/rand/v2"
-
 	"google.golang.org/grpc/balancer"
 	"google.golang.org/grpc/balancer/base"
 	"google.golang.org/grpc/connectivity"
 	"google.golang.org/grpc/internal/balancer/gracefulswitch"
+	"google.golang.org/grpc/internal/envconfig"
 	"google.golang.org/grpc/resolver"
 	"google.golang.org/grpc/serviceconfig"
 )
 
+// PickFirstConfig is a pick first config without shuffling enabled.
+var PickFirstConfig string
+
+func init() {
+	if envconfig.NewPickFirstEnabled {
+		PickFirstConfig = "[{\"pick_first\": {}}]"
+		return
+	}
+	PickFirstConfig = "[{\"pick_first_leaf\": {}}]"
```

Comment:
"If WRR starts delegating to PF, it will loose the health checking capability." - sorry, I don't know what you mean by health checking capability? Aren't the two things that are interested in that Outlier Detection and the health checking from channel? This acts on OOB listeners, but I think it's distinct to pure client side health checking from the channel, right?

Reply:
"Don't merge this PR until the generic health check changes are merged in the new PF. This may take 2+ weeks." - this SGTM as well